src/actions/account.js

```diff
@@ -75,27 +74,33 @@ export const passphraseUsed = data => ({
 /**
  *
  */
-export const secondPassphraseRegistered = ({ secondPassphrase, account, passphrase }) =>
+export const secondPassphraseRegistered = ({
+  secondPassphrase, account, passphrase, callback,
+}) =>
   (dispatch, getState) => {
+    if (account.balance < Fees.setSecondPassphrase) {
```

Comment:
I think this validation should be done outside this thunk and be called before this function.
This function should be only on charge of register the second pass, but I think will be better to avoid that one function manage more than one responsibility. What do you think?

Reply:
I checked with Dominic and implemented checking the balance already on settings page.